packages/ruleset/test/rules/integer-attributes.test.js

```diff
@@ -0,0 +1,443 @@
+  describe('Should not yield errors', () => {
+    it('Clean spec', async () => {
+      const results = await testRule(ruleId, rule, rootDocument);
+      expect(results).toHaveLength(0);
+    });
+    it('Integer schema is in a not', async () => {
+      const testDocument = makeCopy(rootDocument);
+      testDocument.paths['/v1/movies'].post.parameters = [
+        {
+          name: 'filter',
+          in: 'query',
+          schema: {
+            type: 'string',
+            not: {
+              type: 'integer',
+            },
+          },
+        },
+      ];
+
+      const results = await testRule(ruleId, rule, testDocument);
+      expect(results).toHaveLength(0);
+    });
+    it('Integer schema specifies minimum in oneOf schemas', async () => {
+      const testDocument = makeCopy(rootDocument);
+      testDocument.paths['/v1/movies'].post.parameters = [
+        {
+          name: 'ruleTester',
+          in: 'query',
+          schema: {
+            $ref: '#/components/schemas/RuleTester',
+          },
+        },
+      ];
+
+      testDocument.components.schemas['RuleTester'] = {
+        description: 'Tests minimum field within oneOf',
+        type: 'integer',
+        format: 'int64',
+        maximum: 10,
+        oneOf: [
+          {
+            minimum: 0,
+          },
+          {
+            minimum: 1,
+          },
+          {
+            minimum: 2,
+          },
+        ],
+        example: 7,
+      };
+
+      const results = await testRule(ruleId, rule, testDocument);
+      expect(results).toHaveLength(0);
+    });
+
+    it('Integer schema specifies fields in allOf schemas', async () => {
+      const testDocument = makeCopy(rootDocument);
+      testDocument.paths['/v1/movies'].post.parameters = [
+        {
+          name: 'ruleTester1',
+          in: 'query',
+          schema: {
+            $ref: '#/components/schemas/RuleTester1',
+          },
+        },
+        {
+          name: 'ruleTester2',
+          in: 'query',
+          schema: {
+            $ref: '#/components/schemas/RuleTester2',
+          },
+        },
+      ];
+
+      testDocument.components.schemas['RuleTester1'] = {
+        description: 'Tests integer fields within allOf',
+        type: 'integer',
+        allOf: [
+          {
+            minimum: 0,
+          },
+          {
+            maximum: 99,
+          },
+          {
+            format: 'int32',
+          },
+        ],
+        example: 98,
+      };
+      testDocument.components.schemas['RuleTester2'] = {
+        description: 'Tests integer fields within allOf',
+        type: ['integer'],
+        allOf: [
+          {
+            minimum: 1,
+          },
+          {
+            maximum: 38,
+          },
+        ],
+        example: 38,
+      };
+
+      const results = await testRule(ruleId, rule, testDocument);
+      expect(results).toHaveLength(0);
+    });
+
+    it('Response header integer schema has no keywords', async () => {
```

Comment:
Is this called out somewhere as something we shouldn't validate?

Reply:
That test originated in the string-attributes.test.js file and just got carried along when I copy/pasted.  I think your comment also applies to the similar test in string-attributes.test.js.  I'm looking into why we don't expect an error/warning in that scenario. One problem I noticed right off the bat is that the testcase is incorrect in how it tries to add the header definition that should fail validation.
`testDocument.paths['/v1/movies/{movie_id}'].get.responses.headers` should be `testDocument.paths['/v1/movies/{movie_id}'].get.responses[200].headers`.  Although that change didn't result in an error being found either.  Still looking...